cmd/eksctl/main.go

```diff
@@ -18,67 +18,67 @@ import (
-			logger.Debug("ignoring error %q", err.Error())
-		}
-	},
+func addCommands(rootCmd *cobra.Command, flagGrouping *cmdutils.FlagGrouping) {
+	rootCmd.AddCommand(versionCmd(flagGrouping))
+	rootCmd.AddCommand(create.Command(flagGrouping))
+	rootCmd.AddCommand(delete.Command(flagGrouping))
+	rootCmd.AddCommand(get.Command(flagGrouping))
+	rootCmd.AddCommand(update.Command(flagGrouping))
+	rootCmd.AddCommand(scale.Command(flagGrouping))
+	rootCmd.AddCommand(drain.Command(flagGrouping))
+	rootCmd.AddCommand(utils.Command(flagGrouping))
+	rootCmd.AddCommand(completion.Command(rootCmd))
 }
 
-func init() {
+func main() {
+	cobra.EnableCommandSorting = false
 
-	var colorValue string
+	rootCmd := &cobra.Command{
+		Use:   "eksctl [command]",
+		Short: "a CLI for Amazon EKS",
+		Run: func(c *cobra.Command, _ []string) {
+			if err := c.Help(); err != nil {
+				logger.Debug("ignoring cobra error %q", err.Error())
+			}
+		},
+	}
 
 	flagGrouping := cmdutils.NewGrouping()
 
-	addCommands(flagGrouping)
+	addCommands(rootCmd, flagGrouping)
 
 	rootCmd.PersistentFlags().BoolP("help", "h", false, "help for this command")
-	rootCmd.PersistentFlags().StringVarP(&colorValue, "color", "C", "true", "toggle colorized logs (true,false,fabulous)")
 	rootCmd.PersistentFlags().IntVarP(&logger.Level, "verbose", "v", 3, "set log level, use 0 to silence, 4 for debugging and 5 for debugging with AWS debug logging")
 
 	cobra.OnInitialize(func() {
-		// Control colored output
-		color := true
-		fabulous := false
-		switch colorValue {
-		case "false":
-			color = false
-		case "fabulous":
-			color = false
-			fabulous = true
-		}
-		logger.Color = color
-		logger.Fabulous = fabulous
-
-		// Add timestamps for debugging
-		logger.Timestamps = false
-		if logger.Level >= 4 {
-			logger.Timestamps = true
-		}
+		initLogger(rootCmd.PersistentFlags().StringP("color", "C", "true", "toggle colorized logs (valid options: true, false, fabulous)"))
 	})
 
 	rootCmd.SetUsageFunc(flagGrouping.Usage)
-}
 
-func main() {
 	if err := rootCmd.Execute(); err != nil {
-		fmt.Println(err) // outputs cobra errors
+		fmt.Println(err)
 		os.Exit(-1)
 	}
 }
 
-func addCommands(flagGrouping *cmdutils.FlagGrouping) {
-	rootCmd.AddCommand(versionCmd(flagGrouping))
-	rootCmd.AddCommand(create.Command(flagGrouping))
-	rootCmd.AddCommand(delete.Command(flagGrouping))
-	rootCmd.AddCommand(get.Command(flagGrouping))
-	rootCmd.AddCommand(update.Command(flagGrouping))
-	rootCmd.AddCommand(scale.Command(flagGrouping))
-	rootCmd.AddCommand(drain.Command(flagGrouping))
-	rootCmd.AddCommand(utils.Command(flagGrouping))
-	rootCmd.AddCommand(completion.Command(rootCmd))
+func initLogger(colorValue *string) {
+	// Control colored output
+	color := true
+	fabulous := false
+	switch *colorValue {
+	case "false":
+		color = false
+	case "fabulous":
+		color = false
+		fabulous = true
+	}
+	logger.Color = color
+	logger.Fabulous = fabulous
+
+	// Add timestamps for debugging
+	logger.Timestamps = false
```

Comment:
https://github.com/weaveworks/eksctl/blob/5f340977c12398db57ad922d9de4faa43fd64e92/cmd/eksctl/main.go#L80-L83

Very minor nit: I know this part was just extracted from `main`, but this could also be:

```suggestion
	logger.Timestamps = logger.Level >= 4
```

Reply:
good idea, thanks